### PR TITLE
Fix GPU implementation of KeyValueSorter sort with non-zero start

### PR DESCRIPTION
**Pull request by liu15**
…art; also provide warning if non-zero start used with noCopy

### CODE REVIEW

**Comment by liu15** [comment]
This code is currently not exercised in any host code.

**Review by adayton1 — APPROVED** [review]
LGTM

File: src/care/KeyValueSorter_impl.h [hunk 1]
```diff
@@ -60,6 +60,10 @@ CARE_INLINE void sortKeyValueArrays(host_device_ptr<KeyT> & keys,
                                     const size_t start, const size_t len,
                                     const bool noCopy)
 {
+   if (noCopy && start > 0) {
```
[the remark this answers is not in the document]
Comment (liu15) [reply src/care/KeyValueSorter_impl.h:63]: Would you prefer changing the API or using a temp variable?

[the remark this answers is not in the document]
Comment (robinson96) [reply src/care/KeyValueSorter_impl.h:63]: I was thinking a temp variable.

[the remark this answers is not in the document]
Comment (liu15) [reply src/care/KeyValueSorter_impl.h:64]: Does this seem reasonable?

[the remark this answers is not in the document]
Comment (adayton1) [reply src/care/KeyValueSorter_impl.h:64]: Looks reasonable to me.

**Review by adayton1 — APPROVED** [review]
LGTM

**Review by robinson96 — APPROVED** [review]
LGTM

**Pull request merged**